verifier/sources/cpu/layout7/stark_verifier.move

```diff
@@ -0,0 +1,1073 @@
+    // 325
+    const MM_OFFSET_SIZE: u64 = 0x145;
+    // 601
+    const MM_OODS_ALPHA: u64 = 0x259;
+    // 8
+    const MM_OODS_COMMITMENT: u64 = 0x8;
+    // 553
+    const MM_OODS_EVAL_POINTS: u64 = 0x229;
+    // 351
+    const MM_OODS_POINT: u64 = 0x15f;
+    // 359
+    const MM_OODS_VALUES: u64 = 0x167;
+    // 3
+    const MM_PROOF_OF_WORK_BITS: u64 = 0x3;
+    // 5
+    const MM_PUBLIC_INPUT_PTR: u64 = 0x5;
+    // 334
+    const MM_RANGE_CHECK16__PERM__INTERACTION_ELM: u64 = 0x14e;
+    // 337
+    const MM_RANGE_CHECK_MAX: u64 = 0x151;
+    // 336
+    const MM_RANGE_CHECK_MIN: u64 = 0x150;
+    // 6
+    const MM_TRACE_COMMITMENT: u64 = 0x6;
+    // 350
+    const MM_TRACE_GENERATOR: u64 = 0x15e;
+    // 324
+    const MM_TRACE_LENGTH: u64 = 0x144;
+    // 602
+    const MM_TRACE_QUERY_RESPONSES: u64 = 0x25a;
+    // 12
+    const N_COLUMNS_IN_MASK: u64 = 0xc;
+    // 9
+    const N_COLUMNS_IN_TRACE0: u64 = 0x9;
+    // 3
+    const N_COLUMNS_IN_TRACE1: u64 = 0x3;
+    // 6
+    const N_INTERACTION_ELEMENTS: u64 = 0x6;
+    // MASK_SIZE + CONSTRAINTS_DEGREE_BOUND
+    const N_OODS_VALUES: u64 = 0xc2;
+    // 6
+    const OFFSET_EXECUTION_BEGIN_ADDR: u64 = 0x6;
+    // 7
+    const OFFSET_EXECUTION_STOP_PTR: u64 = 0x7;
+    // 3
+    const OFFSET_LAYOUT_CODE: u64 = 0x3;
+    // 0
+    const OFFSET_LOG_N_STEPS: u64 = 0x0;
+    // 20
+    const OFFSET_N_PUBLIC_MEMORY_PAGES: u64 = 0x14;
+    // 4
+    const OFFSET_PROGRAM_BEGIN_ADDR: u64 = 0x4;
+    // 5
+    const OFFSET_PROGRAM_STOP_PTR: u64 = 0x5;
+    // 21
+    const OFFSET_PUBLIC_MEMORY: u64 = 0x15;
+    // 18
+    const OFFSET_PUBLIC_MEMORY_PADDING_ADDR: u64 = 0x12;
+    // 2
+    const OFFSET_RC_MAX: u64 = 0x2;
+    // 1
+    const OFFSET_RC_MIN: u64 = 0x1;
+    // 5
+    const PROOF_PARAMS_FRI_STEPS_OFFSET: u64 = 0x5;
+    // 4
+    const PROOF_PARAMS_N_FRI_STEPS_OFFSET: u64 = 0x4;
+    // 16
+    const PUBLIC_MEMORY_STEP: u256 = 0x10;
+    // 0
+    const REGULAR_PAGE: u256 = 0x0;
+    // End of generating constants!
+
+    friend verifier_addr::gps_statement_verifier;
+    friend verifier_addr::cpu_verifier_7;
+
+    // constants
+    const PROOF_PARAMS_N_QUERIES_OFFSET: u64 = 0;
+    const PROOF_PARAMS_LOG_BLOWUP_FACTOR_OFFSET: u64 = 1;
+    const PROOF_PARAMS_PROOF_OF_WORK_BITS_OFFSET: u64 = 2;
+    const PROOF_PARAMS_FRI_LAST_LAYER_LOG_DEG_BOUND_OFFSET: u64 = 3;
+
+    struct ConstructorConfig has key, copy {
+        // The work required to generate an invalid proof is 2^numSecurityBits.
+        // Typical values: 80-128.
+        num_security_bits: u256,
+
+        // The secuirty of a proof is a composition of bits obtained by PoW and bits obtained by FRI
+        // queries. The verifier requires at least minProofOfWorkBits to be obtained by PoW.
+        // Typical values: 20-30.
+        min_proof_of_work_bits: u256
+    }
+
+    public fun init_stark_verifier(signer: &signer, num_security_bits: u256, min_proof_of_work_bits: u256) {
```

Comment:
public(friend)

Reply:
this function is used for testing and script

Reply:
then restrict who can call this fuction

Reply:
`public` is the only visibility that satisfies

Reply:
handle by writing logic code